### ISSUE TITLE
MySql source : source binary node transforms to text node 

### ISSUE THREAD

**Issue by DoNotPanicUA** [airbytehq/airbyte#5878]
Found during work on #3932

## Current Behavior
Initial investigation showed that the data type test gets airbyte messages with binary values as text.
As a result, check `isBinary` returns false and the value contains useless data (not even string version).

Note! The issue is not investigated fully. There might be a global issue of binary values. 

## Expected Behavior
Binary values in Airbyte message data should be in Binnary nodes for proper processing.

**Comment by sherifnada** [comment]
just to make sure we are aligned, the desired behavior of the connector is to output JSONs as follows: 

```
{
  "binaryField": "<base64 representation of the binary data>"
}
```

this would allow us to use the [`contentEncoding`](https://json-schema.org/understanding-json-schema/reference/non_json_data.html#id2) directive of json schema. We should separately make tickets to:
1. annotate the type for those fields with `contentEncoding=base64` in our json schema, and 
2. make destinations recognize the encoding and write binary columns
 
So for now, the source should output only base64 string encoding of the binary data, and the destination should be expected to convert it to binary again. @irynakruk @DoNotPanicUA do you think this makes sense?

**Comment by DoNotPanicUA** [comment]
I totally agree that it's the best solution, but I'm not sure that we can make it in one dash. I see a few major challenges here:
- JSON schema for Java primitives needs rework. We found that problem when I was fixing BiqQuery Denormalized Destination. (By the way, @VitaliiMaltsev is working on this improvement for another issue right now)
- The common transformation for destinations should be pretty simple, but we also should properly map with binary data types and make sure we can handle it at our implementation. At the most optimistic scenario, we need to do an investigation of it.

So, I propose to split this task into three milestones:
1. Fix the current state and start processing binary values as decoded strings. (As Irina proposed)
2. Improve our JSON schema logic for the Java sources
3. Make a soft migration of all destinations from Binary-String to Binary-Binary processing.

@sherifnada 
For me, it's hard to estimate what is more important here unblock users or store the data at binary data types. So, we need your final solution here. Shall we focus on fixing the issue and moving to your solution step by step or it's better to focus on the final solution from the begging and fix this issue as part of the big change?

**Comment by sherifnada** [comment]
@DoNotPanicUA this sounds like a great approach. So we will first make sure the source outputs b64 encoding, then change the JSON schema representation to express that the data is binary, then migrate destinations to start handling this correctly. 

I think we should do it step by step like you proposed Andrii.
